File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.2 Coupling Exercise/Before/UserDataService.java
package com.grokkingsoftwarearchitecture.chapter03.couplingexercise.before;

import java.util.Arrays;
import java.util.List;

public class UserDataService {
    public String getUserName(int userId) {
        System.out.println("    [Service] Fetching Name...");
        return "Jane Doe";
    }

    public String getUserEmail(int userId) {
        System.out.println("    [Service] Fetching Email...");
        return "jane.doe@example.com";
    }

    public List<String> getUserOrderIds(int userId) {
        System.out.println("    [Service] Fetching Order IDs...");
        return Arrays.asList("A123", "B456");
    }

    public double getOrderTotal(String orderId) {
        System.out.println("    [Service] Fetching Total for Order " + orderId + "...");
        return 99.95;
    }
}

File: Chapter 03/C#/3.2 Coupling Exercise/Before/UserDataService.cs
using System;
using System.Collections.Generic;

namespace Chapter03.CouplingTest.Before;

public class UserDataService
{
    public string GetUserName(int userId) 
    {
        Console.WriteLine("    [Service] Fetching Name...");
        return "Jane Doe"; 
    }
    
    public string GetUserEmail(int userId) 
    { 
        Console.WriteLine("    [Service] Fetching Email...");
        return "jane.doe@example.com"; 
    }
    
    public List<string> GetUserOrderIds(int userId) 
    { 
        Console.WriteLine("    [Service] Fetching Order IDs...");
        return new List<string> { "A123", "B456" }; 
    }
    
    public decimal GetOrderTotal(string orderId) 
    { 
        Console.WriteLine($"    [Service] Fetching Total for Order {orderId}...");
        return 99.95m; 
    }
}